bin/gpo

```diff
@@ -321,13 +321,17 @@ class gPodderCli(object):
         the podcast has not been subscribed to.
         """
         url = util.normalize_feed_url(original_url)
-        if url is None:
-            self._error(_('Invalid url: %s') % original_url)
-            return None
+
+        # Check if it's an Apple Podcasts link, resolves to podcast feed url if that's the case
+        url = util.parse_apple_podcasts_url(url)
 
         # Check if it's a YouTube channel, user, or playlist and resolves it to its feed if that's the case
         url = youtube.parse_youtube_url(url)
 
+        if url is None:
+            self._error(_('Invalid url: %s') % original_url)
+            return None
+
```

Comment:
Why move this check for None here? It makes no sense to feed None to Apple and youtube parsers and then fail.

Reply:
Depends on how much you trust the lookup endpoint. It may return valid json, with feedUrl as `null`, which .json() should convert to None, or return feedUrl where len(url) < 8, in which case [util.normalize_feed_url() returns None](https://github.com/gpodder/gpodder/blob/master/src/gpodder/util.py#L244), and we call that function on the supplied url (untrusted input & for cleanup). So the check has to occur after parse_apple_podcasts_url().
As to why the sequence is normalize->is_apple?->is_youtube?->is_none, it mirrors the sequence in gtkui/main.py, where the check was already last, after youtube link parsing.

Reply:
Also both parsers have an early escape hatch for None, mostly because re.match throws an error on NoneType; the youtube parser has a test for it too.

Reply:
You could do the check for None in `parse_apple_podcasts_url()`, and return the original URL if it fails. This way the function would also do what it says in the docstring.